### PR TITLE
resizable editor

### PR DESCRIPTION
**Pull request by UmaMoiseenko**
Turned out I was at a pretty outdated branch (from w week ago), had to resolve lots of conflicts. 
After merge had error related to `reducers-test.service`, I left it in module, but you'll need to comment that out to test.
Full disclosure - I did not test it cross browser, I worked on latest Chrome..

### CODE REVIEW

**Comment by kirjs** [comment]
Thanks, skimmed through it and it looks nice, I'm going to get into details later today, 
* please add a couple of screenshots, 
* feel free to get rid of the reducers-test.service.ts
* Let me know if you want a thorough code review, or alternatively I could just merge it in, and fix things myself.

**Comment by UmaMoiseenko** [comment]
Nice:) I'd rather have a thorough review. 

The way resizable rows work now -
- if there is an open tab, the tabs below can be dragged up and down to resize that tab
- any tab resizing the closest expanded sibling above
- bottom editor window should take the available space

I made sure that editors take the width of the parent window, so the vertical tabs that contain editors and task description can be potentially resized as well.
screen shot 2016-12-10 at 11 52 43 pm

File: src/app/codelab/codelab.component.css [hunk 1]
```diff
@@ -62,6 +62,7 @@
 }
 
 .codelab-milestone {
+  display: flex;
```
Comment (kirjs) [inline src/app/codelab/codelab.component.css:65]: this sneaked in after merging, please remove

File: src/app/editors/editors.component.ts [hunk 2]
```diff
@@ -50,4 +59,50 @@ export class EditorsComponent {
...
+  // GETTERS
+  getSizeModel(i) {
+    return this.sizeModels[i] = {
+      height: this.getHeight(i),
+      width: this.getWidth(i)
+    }
+  }
+
+  getClosestCollapsedFile(i) {
+    for( i; i >= 0; i--) {
```
Comment (kirjs) [inline src/app/editors/editors.component.ts:72]: Please use Array.prototype.find instead of the loop to save a few lines.

**Comment by kirjs** [comment]
I tested it locally, the code seems to be pretty good, and it works locally, couple of things before I get into details. 
1) This seems to be solving #20 instead of #19. What I had in mind for #19 was horizontal resizing. 
Now the editor get resized as you type.
2) It would be great to have it as a separate component, this way it would be possible to reuse, and also we won't have resizing logic in other components which would simplify understanding them. 
Maybe use https://github.com/mattlewis92/angular-resizable-element for inspiration (or use the actual component?)
wdyt?

**Comment by UmaMoiseenko** [comment]
I see, I misunderstood the issue. Since #20 is marked as fixed, I think it makes more sense to close this pr :)

**Comment by kirjs** [comment]
I'm closing this one, but let me know if you would be interested in adapting your component for vertical resizing. 
I had a very specific idea in mind which I'll elaborate on in an issue later.

**Pull request closed without merging**